types/coin.go

```diff
@@ -320,13 +323,14 @@ func (coins Coins) Empty() bool {
 
 // Returns the amount of a denom from coins
 func (coins Coins) AmountOf(denom string) Int {
+	lowercaseDenom := strings.ToLower(denom)
```

Comment:
In this function, can we also check the case like we do on line 464? I think that would be preferable to just converting to lowercase - we should be enforcing that all "denoms" are lowercase.

Reply:
`AmountOf` at the moment is just converting to lowercase `denom` and process it (basically implementing real case insensitivity). I couldn't quite get what you would like to see instead maybe you would prefer `AmountOf` to panic if a non-all-lowcase string is encountered?

Reply:
Yes, exactly (instead of converting). Just want to ensure we don't accidentally have mixed-case somewhere in our code, which is implicitly converted in `AmountOf` - better to panic (and find such places if they exist).